cellmlmanip/model.py

```diff
@@ -689,17 +689,23 @@ def get_value(self, symbol):
 
     def get_initial_value(self, symbol):
         """Returns the initial value of the given symbol
+
         :param symbol: Sympy Dummy object of required symbol
         :return: float of initial value
         """
         return float(self.dummy_metadata[symbol].initial_value)
 
     def find_symbols_and_derivatives(self, expression):
-        """ Returns a set containing all symbols and derivatives referenced in an expression. """
-        symbols = set()
+        """ Returns a set containing all symbols and derivatives referenced in an expression or list of expressions.
 
+        :param expression: expression or list of expressions to get symbols for
+        """
+        symbols = set()
+        if isinstance(expression, list):
```

Comment:
This only accepts lists, not sequences in general, so usually a bad idea to check like this

Reply:
Changed it to Iterable

Reply:
hold on hold on, let's just use the code we had before!